py/torch_tensorrt/_Input.py

```diff
@@ -171,6 +172,14 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
 
         self.tensor_domain = Input._parse_tensor_domain(domain)
 
+        if "torch_tensor" in kwargs:
+            self.torch_tensor = kwargs["torch_tensor"]
+        else:
+            if self.shape_mode == Input._ShapeMode.DYNAMIC:
+                self.torch_tensor = self.example_tensor("opt_shape")
+            else:
+                self.torch_tensor = self.example_tensor()
```

Comment:
What if `"torch_tensor"` is provided, but the shape is also Dynamic - would it be an issue if this Tensor had the shape of `min` or `max` instead of `opt`? This could be refactored to override or validate the specified tensor if the shape is dynamic.

Reply:
>  would it be an issue if this Tensor had the shape of min or max instead of opt?
So the usecase you are saying is if the user provides 
`x = torch_tensorrt.Input(min_shape=<>, opt_shape=<>, max_shape=<>, torch_tensor=<random_tensor>)`
This wouldn't be an issue, as we ignore torch_tensor in this case of dynamic shape compilation. https://github.com/pytorch/TensorRT/blob/695dc9b5f4a2a5f22b40416c7c83ed0eddc6c39a/py/torch_tensorrt/dynamo/partitioning/common.py#L48 would generate input tensors for all the modes.
We could maybe pass a warning to users that torch_tensor is not being used in such cases.

Reply:
I see - I believe the feature ask from #2323 is to allow providing tensors for each of `min`, `opt`, and `max`. Could `torch_tensor` be instead either a list/tuple of 3 tensors (`min`, `opt`, `max`) _or_ a single tensor?